src/Interpreters/AsynchronousMetrics.h

```diff
@@ -45,6 +45,7 @@ class AsynchronousMetrics : WithContext
         ContextPtr global_context_,
         int update_period_seconds,
         std::shared_ptr<std::vector<ProtocolServerAdapter>> servers_to_start_before_tables_,
+        std::shared_ptr<std::mutex> servers_lock_,
```

Comment:
Sharing of a mutex doesn't look very nice to me. Maybe it's better to change something. For example, it seems instead of passing `servers_to_start_before_tables_`, `servers_lock_`, `servers_` we can pass a function of type `std::function<std::vector<ProtocolServerMetrics>()>`, where `ProtocolServerMetrics` is defined as following

```
struct ProtocolServerMetrics
{
  String port_name;
  size_t current_threads;
};
```

Thus we will able to turn both the mutex and the list of servers to simple local variables in `Server.cpp`, without necessity to share them with another module.